x-pack/plugins/data_visualizer/public/application/file_data_visualizer/components/json_editor/json_editor.tsx

```diff
@@ -32,29 +26,22 @@ export const JsonEditor: FC<JobEditorProps> = ({
   width = '100%',
   mode = EDITOR_MODE.JSON,
   readOnly = false,
-  syntaxChecking = true,
-  theme = 'textmate',
   onChange = () => {},
 }) => {
   if (mode === EDITOR_MODE.XJSON) {
     value = expandLiteralStrings(value);
   }
 
   return (
-    <EuiCodeEditor
+    <CodeEditor
       value={value}
       width={width}
       height={height}
-      mode={mode}
-      readOnly={readOnly}
-      wrapEnabled={true}
-      showPrintMargin={false}
-      theme={theme}
-      editorProps={{ $blockScrolling: true }}
-      setOptions={{
-        useWorker: syntaxChecking,
+      languageId={mode}
+      options={{
         tabSize: 2,
-        useSoftTabs: true,
+        readOnly,
+        automaticLayout: true,
```

Comment:
Not sure if this prop is related, but I am seeing overflow into the parent container as I resize the window which isn't happening on `main`:

<img width="304" alt="image" src="https://github.com/elastic/kibana/assets/7405507/3df0dd98-0871-485c-9975-374f320edac1">